state/applicationoffers.go

```diff
@@ -330,20 +361,15 @@ func removeApplicationOffersOps(st *State, application string) ([]txn.Op, error)
 	if n == 0 {
 		return []txn.Op{countRefsOp}, nil
 	}
-
-	applicationOffersCollection, closer := st.db().GetCollection(applicationOffersC)
-	defer closer()
-	query := bson.D{{"application-name", application}}
-	var docs []applicationOfferDoc
-	if err := applicationOffersCollection.Find(query).All(&docs); err != nil {
-		return nil, errors.Annotatef(err, "reading application %q offers", application)
+	docs, err := applicationOffersDocs(st, application)
+	if err != nil {
+		return nil, errors.Trace(err)
 	}
-
 	var ops []txn.Op
 	for _, doc := range docs {
 		ops = append(ops, txn.Op{
 			C:      applicationOffersC,
-			Id:     doc.OfferName,
+			Id:     doc.DocID,
```

Comment:
@wallyworld I am pretty sure that this change is correct - what do you think?

Reply:
From memory, the doc id is set to the OfferName so it works ether way. Your way is better though